main.go

```diff
@@ -97,22 +100,41 @@ func workLoop(ctx context.Context, cfg *config.Config, client agent.AgentClient)
 			}
 
 		case *agent.ServerMessage_State:
-			for _, agent := range payload.State.AgentProcesses {
-				switch agent.Type {
-				case inventory.AgentType_MYSQLD_EXPORTER:
-					if err := supervisor.StartMySQLdExporter(agent.Args, agent.Env); err != nil {
-						l.Error(err)
-					}
-				default:
-					l.Warnf("Got unhandled agent type %s (%d), ignoring.", agent.Type, agent.Type)
+
+			var agentProcessesStates []*agent.SetStateResponse_AgentProcess
+
+			for _, agentProcess := range payload.State.AgentProcesses {
+
+				params := &runner.AgentParams{
+					AgentId: agentProcess.AgentId,
+					Type:    agentProcess.Type,
+					Args:    agentProcess.Args,
+					Configs: agentProcess.Configs,
+					Env:     agentProcess.Env,
+					Port:    9172,
```

Comment:
Port should be allocated dynamically. Something like https://github.com/percona/pmm-managed/blob/43842b2ccd7f72d5935ef2e3297e3cdd1fa26d60/utils/ports/registry.go